src/Setup/UpdateVersions.purs

```diff
@@ -34,141 +34,171 @@ import Node.FS.Sync (writeTextFile)
 updateVersions :: Aff Unit
 updateVersions = do
   versions <- for Tool.allTools \tool -> do
     delay (Milliseconds 500.0)
-    version <- fetchLatestReleaseVersion tool
-    pure $ Tuple tool version
+    versionRec <- fetchLatestReleaseVersion tool
+    pure $ Tuple tool versionRec
 
-  let
-    insert obj (Tuple tool version) =
-      Tool.name tool := Version.showVersion version ~> obj
-
-  liftEffect $ writeVersionsFile $ foldl insert jsonEmptyObject versions
+  updateV1File versions
+  updateV2File versions
   where
-  versionsFilePath :: FilePath
-  versionsFilePath = "./dist/versions.json"
-
-  writeVersionsFile :: Json -> Effect Unit
-  writeVersionsFile = writeTextFile UTF8 versionsFilePath <<< (_ <> "\n") <<< stringifyWithIndent 2
+  updateV1File versions = liftEffect do
+    let V1FileSchema { localFile, encode } = version1
+    writeVersionsFile localFile
+      $ encode
+      $ Map.fromFoldable
+      $ map (map _.latest) versions
+
+  updateV2File versions = liftEffect do
+    let V2FileSchema { localFile, encode } = version2
+    writeVersionsFile localFile
+      $ encode
+      $ Map.fromFoldable versions
+
+  writeVersionsFile :: FilePath -> Json -> Effect Unit
+  writeVersionsFile path = writeTextFile UTF8 path <<< (_ <> "\n") <<< stringifyWithIndent 2
 
 -- | Find the latest release version for a given tool. Prefers explicit releases
 -- | as listed in GitHub releases, but for tools which don't support GitHub
 -- | releases, falls back to the highest valid semantic version tag for the tool.
-fetchLatestReleaseVersion :: Tool -> Aff Version
+fetchLatestReleaseVersion :: Tool -> Aff { latest :: Version, unstable :: Version }
 fetchLatestReleaseVersion tool = Tool.repository tool # case tool of
   PureScript -> fetchFromGitHubReleases
   Spago -> fetchFromGitHubReleases
   Psa -> fetchFromGitHubTags
   PursTidy -> fetchFromGitHubTags
   Zephyr -> fetchFromGitHubReleases
-  where
-  -- TODO: These functions really ought to be in ExceptT to avoid all the
-  -- nested branches.
-  fetchFromGitHubReleases repo = recover do
-    page <- liftEffect (Ref.new 1)
-    untilJust do
-      versions <- liftEffect (Ref.read page) >>= toolVersions repo
-      case versions of
-        Just versions' -> do
-          let version = Array.find (not <<< Version.isPreRelease) versions'
-          when (isNothing version) do
+
+-- TODO: These functions really ought to be in ExceptT to avoid all the
+-- nested branches.
+fetchFromGitHubReleases :: Tool.ToolRepository -> Aff { latest :: Version, unstable :: Version }
+fetchFromGitHubReleases repo = recover do
+  page <- liftEffect (Ref.new 1)
+  untilBothVersionsFound \firstUnstableVersion -> do
+    versions <- liftEffect (Ref.read page) >>= toolVersions repo
+    case versions of
+      Just versions' -> do
+        let
+          unstable = firstUnstableVersion <|> Array.head versions'
+          latest = Array.find (not <<< Version.isPreRelease) versions'
```

Comment:
This really ought to check for build metadata, too. We can check that the `Version.buildMetadata` list is empty.

https://github.com/hdgarrood/purescript-versions/blob/9799321bfa516b05d68bea8da7b706219e87edd8/src/Data/Version.purs#L85-L86